docs/design-documents/README.md

```diff
@@ -0,0 +1,40 @@
+# Design Process for Mbed-OS
+This document defines the software design process for Mbed-OS. Anyone developing for Mbed-OS should use this process to submit the design to get it reviewed from the broader Mbed-OS developer community.
+
+## Design documents location and organization
+* It's suggested to write one design document per change, which could be a new feature, defect fixes, tools changes etc. Do not try to address multiple changes in the same document.
+* The design document will follow the format represented in [Design Document Template](https://github.com/ARMmbed/mbed-os/blob/master/docs/design-documents/design_template.md)
+
+* For each design submission, create a folder under *https://github.com/ARMmbed/mbed-os/docs/design-documents/{AREA OR COMPONENT}*. The area/component would be Drivers, Platform, Connectivity, Security, Peripherals etc. 
+  If your implementation is creating new APIs, you should try to match the folder name directly to the new API, otherwise 
+  use the title of the feature as the folder name. 
+  For example, if your implementation is adding new EMAC API, you may name the folder as **New_EMAC_API** or if you are creating a new Serial Driver for Mbed-OS you may name the folder as **Serial_driver_for_Mbed-os**.
+* Your design document itself should also be named with the same name of the as the folder, so that it serves
+  as your master design document, if you happen to use other .md files under same folder as annexures or reference.
+* The diagrams you are using may exist in the sub-folder *diagrams* under the new folder you have created.
+  See [Example Design Document](https://github.com/ARMmbed/mbed-os/blob/master/docs/design-documents/example_feature_design/example_feature_design.md) for example design document.
+
+### Using diagrams in Design documents
+Its highly encouraged to create diagrams to depict your software design. The [Design Document Template]((https://github.com/ARMmbed/mbed-os/blob/master/docs/design-documents/design_template.md) provided here uses diagrams generated using draw.io. draw.io generates XMLs for each diagram and those corresponding XMLs are also provided under the diagrams folder, and you are free to use that as your reference for generating your own diagrams. And, when you are generating your own diagrams it would be better to provide the metadata/other files(e.g, if you are using draw.io provide XMLs from draw.io) required for future editing of those diagrams.
+
+## Design Process
+* Developer generates a design document located under *https://github.com/ARMmbed/mbed-os/docs/design-documents/{AREA OR COMPONENT}* and generates a pull-request to Mbed-OS.
```

Comment:
Would this be rather : first github issue to discuss the idea (gather feedback) - is it worth my and yours time? Then take time to write it down and create PR with the design specifications

Reply:
Its ok for developers to gather feedback on some idea if they want, which I think is already happening. But when they have decided to go ahead implement the idea a design document could be generated to capture more details and for later reference.

Reply:
> Its ok for developers to gather feedback on some idea if they want, which I think is already 
happening.

I was after if someone already put effort to write down and it gets rejected - I have seen it with features - maybe my question should have  been : when is design document approved and integrated (as well when its rejected and closed) ? How to maximize the success of the design document. I believe ideally it would be if issue gets enough traction and +1 from involved parties, design document would be approved and later implemented.